io/disk/ssd/nvmetest.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+NVM-Express user space tooling for Linux, which handles NVMe devices.
+This Suite creates and formats a namespace, reads and writes on it
+using nvme cli.
+"""
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process
+import avocado.utils.git as git
+
+
+class NVMeTest(Test):
+
+    """
+    NVM-Express user space tooling for Linux, which handles NVMe devices.
+
+    :param device: Name of the nvme device
+    """
+
+    def setUp(self):
+        """
+        Build 'nvme-cli' and setup the device.
+        """
+        nvme_path = os.path.join(self.srcdir, 'nvme')
+        nvme_link = 'https://github.com/linux-nvme/nvme-cli.git'
+        git.get_repo(nvme_link, destination_dir=nvme_path)
+        build.make(nvme_path)
+        build.make(nvme_path, extra_args='install')
+        self.device = self.params.get('device', default='/dev/nvme0')
+        # cmd = "ls -l %sn* | sort | awk '{print $NF}'" % self.device
+        # self.id_ns = process.system_output(cmd, shell=True).strip('\n'):
+        # if 'No such file or directory' in self.id_ns:
+        #     self.id_ns = self.device + 'n1'
+        self.id_ns = self.create_namespace()
+        self.format_namespace()
+        cmd = "nvme id-ns %s | grep 'in use' | awk '{print $5}' | \
+            awk -F':' '{print $NF}'" % self.id_ns
+        self.format_size = process.system_output(cmd, shell=True).strip('\n')
+        self.format_size = pow(2, int(self.format_size))
+        cmd = "nvme id-ns %s | grep 'in use' | awk '{print $2}'" % self.id_ns
+        self.lba = process.system_output(cmd, shell=True).strip('\n')
+        print self.device, self.id_ns, self.lba, self.format_size
+
+    def create_namespace(self):
+        """
+        Creates namespace on the device.
+        """
+        cmd = "nvme show-regs %s | grep version | awk '{print $NF}'" \
+            % self.device
+        if int(process.system_output(cmd, shell=True).strip('\n')) < 10200:
+            return self.device + 'n1'
```

Comment:
That is because I am yet to implement the determination of actual self.id_ns value, for which I need a latest adapter.
Since I do not have that yet, this is a placeholder.
I have mentioned this in TODO